src/local-ssh/telemetryService.ts

```diff
@@ -0,0 +1,331 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Gitpod. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as os from 'os';
+import { Analytics, AnalyticsSettings } from '@segment/analytics-node';
+import { cloneAndChange, escapeRegExpCharacters, mixin } from '../common/utils';
```

Comment:
I wonder there is no way to pull common code from here under `common/telemetry` to reuse between extension and proxy code?

Reply:
I do want to do it https://github.com/gitpod-io/gitpod-vscode-desktop/pull/65#discussion_r1197391854 , but we should remove `import 'vscode'` otherwise `proxy.ts` will not work. And it seems the code is not that good to do it.

Reply:
yeah, I mean pull super class or something, leave template methods where vscode is needed?

Reply:
PR https://github.com/gitpod-io/gitpod-vscode-desktop/pull/72 to bring same code into common. But don't know if @jeanp413  like it or not